app/src/main/java/com/github/se/polyfit/ui/components/IngredientList.kt

```diff
@@ -0,0 +1,100 @@
+package com.github.se.polyfit.ui.components
+
+import android.util.Log
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.unit.dp
+import com.github.se.polyfit.ui.screen.Ingredient
+
+@OptIn(ExperimentalLayoutApi::class)
+@Composable
+fun IngredientList(
+    it: PaddingValues,
+    ingredients: List<Ingredient>,
+    potentialIngredients: List<Ingredient>,
+    onAddIngredient: (Int) -> Unit,
+) {
+
+  val initialSuggestions = 3
+  val potentialIndex = remember {
+    // equivalent to min(3, potentialIngredients.size)
+    mutableIntStateOf(initialSuggestions.coerceAtMost(potentialIngredients.size))
+  }
+  Column(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(it)
+              .verticalScroll(rememberScrollState())
+              .testTag("IngredientsList"),
+      verticalArrangement = Arrangement.Top,
+      horizontalAlignment = Alignment.CenterHorizontally) {
+        if (ingredients.isEmpty() && potentialIngredients.isEmpty()) {
+          Text(
+              "No ingredients added yet",
+              modifier = Modifier.fillMaxSize().padding(16.dp, 0.dp).testTag("NoIngredients"),
+              color = MaterialTheme.colorScheme.secondary,
+              fontSize = MaterialTheme.typography.headlineSmall.fontSize)
+        } else {
+          FlowRow(
+              horizontalArrangement = Arrangement.Start, verticalArrangement = Arrangement.Top) {
+                ingredients.forEach { ingredient ->
+                  GradientButton(
+                      text = "${ingredient.name} ${ingredient.quantity}${ingredient.unit}",
```

Comment:
Consider adding a space between the quantity and the unit to make the text look cleaner.

Reply:
Yeah ill also update this with a helper in `Ingredient` class when doing the data integration